File: 0443-string-compression/0443-string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int l=0, r=0, length=0, count=0;
        int i=0;
        while(r<=chars.size()&&l<chars.size()){
            if(r==chars.size()||chars[l]!=chars[r]){
                string s;
                s.push_back(chars[l]);
                if(count!=1)    s += to_string(count);
                // cout<<s<<endl;
                for(int c=0;c<s.size();c++)  chars[i+c]=s[c];
                i+=s.size();
                length+= s.size();
                l=r;    count=0;
            }else{
                count++;
                r++;
            } 
        }
        return length;
    }
};

File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        left = 0
        ans_ptr = 0
        while left < len(chars):
            right = left
            while right < len(chars) and chars[right]==chars[left]:
                right += 1
            
            chars[ans_ptr] = chars[left]
            if right-left > 1 and ans_ptr+1 < len(chars):
                m = str(right-left)
                for c in m:
                    chars[ans_ptr+1] = c
                    ans_ptr += 1
            ans_ptr += 1
            
            if right < ans_ptr:
                left = ans_ptr
            else:
                left = right

        return ans_ptr
            